src/builder/pod/mod.rs

```diff
@@ -390,31 +499,40 @@ impl PodBuilder {
             image_pull_secrets: self.image_pull_secrets.clone(),
             restart_policy: self.restart_policy.clone(),
             ..PodSpec::default()
+        };
+
+        // We don't hard error here, because if we do, the StatefulSet (for
+        // example) doesn't show up at all. Instead users then need to comb
+        // through the logs to find the error. That's why we opted to just
+        // throw a warning which will get displayed in the Kubernetes
+        // status. Additionally the Statefulset will have events describing the
+        // actual problem.
+
```

Comment:
I would prefer something like this, as this is more compact (same in cluster_resources)
```
        pod_spec
            .check_resource_requirement(ResourceRequirementsType::Limits, "cpu")
            .unwrap_or_else(|err| warn!("{err}"));
        pod_spec
            .check_resource_requirement(ResourceRequirementsType::Limits, "memory")
            .unwrap_or_else(|err| warn!("{err}"));
        pod_spec
            .check_limit_to_request_ratio(&ComputeResource::Cpu, LIMIT_REQUEST_RATIO_CPU)
            .unwrap_or_else(|err| warn!("{err}"));
        pod_spec
            .check_limit_to_request_ratio(&ComputeResource::Memory, LIMIT_REQUEST_RATIO_MEMORY)
            .unwrap_or_else(|err| warn!("{err}"));
```

Reply:
Changed in f5a7a7c

Reply:
Please also for cluster_resources code :)

Reply:
Updated `cluster_resources` in e57a117